packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -38,31 +70,79 @@ const ContextMenu: React.FC<{ onOpenChange?(open: boolean): void }> = (props) =>
   );
 
   return (
-    <MenuPrimitive.Root open={open} onOpenChange={handleOpenChange}>
+    <MenuPrimitive.Root open={open} onOpenChange={handleOpenChange} dir={dir}>
       <ContextMenuProvider open={open} onOpenChange={handleOpenChange}>
         {children}
       </ContextMenuProvider>
     </MenuPrimitive.Root>
   );
 };
 
+/* -----------------------------------------------------------------------------------------------*/
+
+type ContextMenuSubOwnProps = {
+  open?: boolean;
+  onOpenChange?(open: boolean): void;
```

Comment:
hmmm perhaps we shouldn't be spreading on to the component here? I worry about these chains of prop inheritance. If a required prop is added to `MenuPrimitive.Sub` for an internal reason, it would unintentionally make its way up to `ContextMenu`.

Reply:
> hmmm perhaps we shouldn't be spreading on to the component here?

Ah yes, I think you're right here. It's just a provider so doesn't need to spread props.

> I worry about these chains of prop inheritance. If a required prop is added to `MenuPrimitive.Sub` for an internal reason, it would unintentionally make its way up to `ContextMenu`.

I think that would be intentional if we're inheriting. That's the expectation in my mind. I'm curious why you think that would be an issue? I find it's usually a lot less overhead to inherit and omit than to create manual duplications of a component's prop types at every level.

Interested in your reasoning on this one though. @benoitgrelard what are your thoughts here? Obviously one benefit of avoiding inheritance is TS perf 😅

Reply:
I think inheriting in general makes sense when we spread props onto another component, it makes sense to me to describe the props of the new based on that, and omit the ones that we hardcode.

Reply:
Agree on the inheritance when spreading. My original bad feeling was that we would merge just because the types happen to line up. I don't think that was the nature of the feedback though 👍 

I'll remove the spread from here.